### PR TITLE
Letters of Support step for users with disabled JS

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/controllers/form/support_letters_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Form::SupportLettersController < Form::BaseController
+  before_action :load_letter, only: :destroy
+
+  def create
```
Comment (matthewford) [inline app/controllers/form/support_letters_controller.rb]: Assignment Branch Condition size for create is too high. [16.16/15]

File: app/controllers/form/support_letters_controller.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @support_letter = @form_answer.support_letters.new(
+      support_letter_params.merge({
```
Comment (matthewford) [inline app/controllers/form/support_letters_controller.rb]: Redundant curly braces around a hash parameter.

File: app/controllers/form/support_letters_controller.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        user_id: current_user.id,
```
Comment (matthewford) [inline app/controllers/form/support_letters_controller.rb]: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: app/controllers/form/support_letters_controller.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        manual: true
+      })
```
Comment (matthewford) [inline app/controllers/form/support_letters_controller.rb]: Indent the right brace the same as the first position after the preceding left parenthesis.

File: app/models/support_letter.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@ class SupportLetter < ActiveRecord::Base
     belongs_to :form_answer
     belongs_to :user
 
-    has_one :support_letter_attachment
+    has_one :support_letter_attachment, autosave: true, dependent: :destroy
```
Comment (matthewford) [inline app/models/support_letter.rb]: Inconsistent indentation detected.

File: app/models/support_letter.rb [hunk 6]
```diff
@@ -13,7 +13,8 @@ class SupportLetter < ActiveRecord::Base
               :user,
               :form_answer,
               :relationship_to_nominee, presence: true
-    validates :attachment, presence: true, if: "self.manual?"
+    validates :attachment, presence: true, if: proc { manual? && !support_letter_attachment }
```
Comment (matthewford) [inline app/models/support_letter.rb]: Inconsistent indentation detected.

File: app/models/support_letter.rb [hunk 7]
```diff
@@ -13,7 +13,8 @@ class SupportLetter < ActiveRecord::Base
...
+    validates :support_letter_attachment, presence: true, if: proc { manual? }
```
Comment (matthewford) [inline app/models/support_letter.rb]: Inconsistent indentation detected.

File: spec/features/users/form_answers/letters_of_support_spec.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  let(:form_answer) { create :form_answer, :promotion, user: user }
+  let!(:settings) { create :settings, :submission_deadlines }
+
+  before do
+    login_as user
+    visit form_form_answer_supporters_path(form_answer)
+  end
+
+  it "should be able to create supporter" do
+    save_and_open_page
```
Comment (matthewford) [inline spec/features/users/form_answers/letters_of_support_spec.rb]: Remove debugger entry point `save_and_open_page`.

**Comment by RuslanHamidullin** [comment]
@dreamfall looks fine for me :+1:

**Comment by RuslanHamidullin** [comment]
Awaiting for travis

**Pull request merged**